providers/base/bin/snap_update_test.py

```diff
@@ -148,11 +151,25 @@ def snap_refresh(self):
                 self.name, original_revision, self.revision
             )
         )
-        response = self.snapd.refresh(
-            self.name,
-            channel=self.snap_info.tracking_channel,
-            revision=self.revision,
-        )
+        try:
+            response = self.snapd.refresh(
+                self.name,
+                channel=self.snap_info.tracking_channel,
+                revision=self.revision,
+            )
+        except (SnapdRequestError, AsyncException) as exc:
+            checkbox_session_dir = os.getenv("PLAINBOX_SESSION_SHARE")
+            if checkbox_session_dir:
+                result = {
+                    "outcome": "fail",
+                    "comments": exc.message,
```

Comment:
Consider changing this to:
```suggestion
                    "comments": "Marking the test as failed because it raised the following:" + str(exc),
```